sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala

```diff
@@ -122,24 +123,45 @@ class DataSourceScanExecRedactionSuite extends DataSourceScanRedactionTest {
   test("SPARK-31793: FileSourceScanExec metadata should contain limited file paths") {
     withTempPath { path =>
       val dir = path.getCanonicalPath
+
+      // create a sub-directory with long name so that each root path will always exceed the limit
```

Comment:
Let's just ensure we test on path truncation; other cases should be tested against UtilsSuite.